hphp/hack/test/typecheck/instance_property_as_class_constant.php

```diff
@@ -0,0 +1,18 @@
+<?hh // strict
+/**
+ * Copyright (c) 2014, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the "hack" directory of this source tree.
+ *
+ *
+ */
+
+class X {                                                                                  
+    public ?string $foo;                                                                   
+}                                                                                          
+                                                                                           
+function main(): void {                                                                    
+    var_dump(X::foo);                                                                      
+}     
```

Comment:
Small nit: you've got trailing whitespace here. Could you remove it?

```suggestion
class X {
    public ?string $foo;
}

function main(): void {
    var_dump(X::foo);
}

```